.github/workflows/test-recent-mainnet-block.yml

```diff
@@ -33,11 +33,9 @@ jobs:
 
       - name: Run test
         uses: actions-rs/cargo@v1
```

Comment:
FYI, https://github.com/actions-rs has been archived last year and the actions repos haven't been updated for quite a while. I think it'd be nice to replace them, but maybe best to do it in a separate PR - I could propose it some time this week if you like the idea.

Reply:
As we use this across all of our GitHub actions, it would be good to do it in a follow-up task.

I created one in our backlog: https://github.com/NomicFoundation/edr/issues/642

Reply:
On a somewhat related note, do we need `actions-rs/cargo` (cargo specifically) at all? It seem to be just running `cargo` and we don't seem to need any cross-compilation or other target specs in general (which could be noisy, I guess), so this might be a regular `run` step, instead.